### PR TITLE
Added deploy script for tt02/prod post release publish

### PR DESCRIPTION
**Pull request by acn-dgopa**
## Related Issue(s)
- #{issue number}

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
+name: Deploy to TT02 and Production
```
Comment (Alxandr) [inline .github/workflows/deploy.yml:1]:
```suggestion
run-name: Deploy version ${{ github.release.name }} to TT02 and Production
```

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            --name ${{ vars.CONTAINER_APP_NAME }} \
+            --container-name ${{ env.CONTAINER_APP_CONTAINER_NAME }} \
+            --resource-group ${{ vars.CONTAINER_APP_RESOURCE_GROUP_NAME }} \
+            --image ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+    - name: logout
+      run: >
+        az logout
+
+  deploy-prod:
+    environment: PROD
```
Comment (Alxandr) [inline .github/workflows/deploy.yml:57]: This job should run after deploy-tt02, so it needs a "needs" clause

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+jobs:
+  deploy-tt02:
+    environment: TT02
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+      packages: read
+    steps:
+    - name: Log in to the Container registry
```
Comment (Alxandr) [inline .github/workflows/deploy.yml:29]: Why are you logging in to the container registry?

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      with:
+        inlineScript: >
+          az config set extension.use_dynamic_install=yes_without_prompt
+          
+          az containerapp update \
+            --name ${{ vars.CONTAINER_APP_NAME }} \
+            --container-name ${{ env.CONTAINER_APP_CONTAINER_NAME }} \
+            --resource-group ${{ vars.CONTAINER_APP_RESOURCE_GROUP_NAME }} \
+            --image ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+    - name: logout
```
Comment (Alxandr) [inline .github/workflows/deploy.yml:52]: Also why?

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (acn-dgopa) [reply .github/workflows/deploy.yml:29]: it was required when we were using azure container registry (private) at the start and then moved to the github package registry which is now public. I failed to remove it after refactoring

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (acn-dgopa) [reply .github/workflows/deploy.yml:39]: fixed

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (acn-dgopa) [reply .github/workflows/deploy.yml]: removed

File: .github/workflows/deploy.yml (hunk 4 above)
Comment (acn-dgopa) [reply .github/workflows/deploy.yml]: removed

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+    - name: Deploy to containerapp
+      uses: azure/CLI@v1
+      with:
+        inlineScript: >
+          az config set extension.use_dynamic_install=yes_without_prompt
+          
+          az containerapp update \
+            --name ${{ vars.CONTAINER_APP_NAME }} \
```
Comment (Alxandr) [inline .github/workflows/deploy.yml:33]: It's generally recommended to not use string interpolation in scripts in github actions. Instead you should pass in environment variables and refer to them in the script. This has several advantages, several of which are safety related.

File: .github/workflows/deploy.yml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+      uses: azure/CLI@v1
+      with:
+        inlineScript: >
+            az config set extension.use_dynamic_install=yes_without_prompt
+            
+            az containerapp update \
+            --name ${{ vars.CONTAINER_APP_NAME }} \
+            --container-name ${{ env.CONTAINER_APP_CONTAINER_NAME }} \
+            --resource-group ${{ vars.CONTAINER_APP_RESOURCE_GROUP_NAME }} \
+            --image ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
```
Comment (Alxandr) [inline .github/workflows/deploy.yml:63]: nit: Git generally prefers files ending in newline.

File: .github/workflows/release-drafter-ca.yml [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+    
+    steps:
+      - name: Get latest version for function app
+        uses: oprypin/find-latest-tag@v1
+        continue-on-error: true
+        with:
+          repository: ${{ env.REPOSITORY }}
+          releases-only: true
+          prefix: ${{ env.RELEASEPREFIX }}${{ env.VERSIONPREFIX }}  # Other kinds of tags are irrelevant;
+        id: auditlogca_version
```
Comment (Alxandr) [inline .github/workflows/release-drafter-ca.yml:30]: Please have a newline between steps here. It makes it much less likely to overlook in review.

File: .github/workflows/release-drafter-ca.yml [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+      - name: setup current year
+        id: get_year
+        run: |
+          echo "currentyear=$(date +%Y)" >> $GITHUB_ENV
+
+      - name: print status
+        run: "echo Release could not be found. Request failed with status '${{ steps.get_latest_release_version.outputs.status }}'"
+      
+      - name: split release version f.ex v2023.1 will be split into output0 v2023 and output1 1
```
Comment (Alxandr) [inline .github/workflows/release-drafter-ca.yml:39]: This seems like a very convoluted way to do some simple string manipulation. Also, I'd like to have a discussion regarding this versioning scheme.

File: .github/workflows/release-drafter-ca.yml [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+          if [ ${{ steps.auditlogca_version.outcome }} = 'success' ]
+          then
+            echo "RELEASE=${{ env.RELEASEPREFIX }}${{ steps.format_version.outputs.RELEASEVERSION }}" >> $GITHUB_OUTPUT
+          else
+            echo "RELEASE=${{ env.RELEASEPREFIX }}${{ env.VERSIONPREFIX }}${{ env.currentyear }}.1" >> $GITHUB_OUTPUT
+          fi
+    outputs:
+      RELEASEVERSION: ${{ steps.releaseversion.outputs.RELEASE }}
+    
+  tag:
```
Comment (Alxandr) [inline .github/workflows/release-drafter-ca.yml:88]: Why is this a separate job?

File: .github/workflows/release-drafter-ca.yml [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+        uses: battila7/get-version-action@v2
+      - name: Add Version tag to Docker Image
+        uses: shrink/actions-docker-registry-tag@v4
+        with:
+          registry: ${{ env.REGISTRY }}
+          repository: '${{ env.REPOSITORY }}'
+          target: '${{ steps.get_imagetag.outputs.imagetag }}'
+          tags: ${{ needs.getlatestrelease.outputs.RELEASEVERSION }}
+
+  update_release_draft:
```
Comment (Alxandr) [inline .github/workflows/release-drafter-ca.yml:132]: Why is this a separate job?

File: .github/workflows/release-drafter-ca.yml [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    steps:
+      # Drafts your next Release notes as Pull Requests are merged into "master"
+      - uses: release-drafter/release-drafter@v6
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag: ${{ needs.getlatestrelease.outputs.RELEASEVERSION }}
+          version: ${{ needs.getlatestrelease.outputs.RELEASEVERSION }}
```
Comment (Alxandr) [inline .github/workflows/release-drafter-ca.yml:147]: Nit: Git generally prefers files ending in newline.

File: .github/workflows/release-drafter-fa.yml [hunk 12]
```diff
@@ -0,0 +1,110 @@
+name: Schedule a release and draft notes for function app
```
Comment (Alxandr) [inline .github/workflows/release-drafter-fa.yml:1]: Basically the same comments as with the previous file. Including the newline at the end.

File: .github/workflows/deploy-after-release.yml [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+      - uses: ./.github/actions/deploy
+        name: Deploy
+        with:
+          image-tag: ${{ github.sha }}
+          resource-group: ${{ vars.CONTAINER_APP_RESOURCE_GROUP_NAME }}
+          container-app: ${{ vars.CONTAINER_APP_NAME }}
+          function-app: ${{ vars.AZURE_FUNCTIONAPP_NAME }}
+          function-app-path: ./artifacts/function-app
+      
+      - name: Add Version tag to Docker Image
```
Comment (Alxandr) [inline .github/workflows/deploy-after-release.yml:79]: This should probably just be moved to the build step. It at least has to happen before the deploy.

File: .github/workflows/create-release-draft.yml [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+
+    permissions:
+      packages: write
+      contents: write  # for release-drafter/release-drafter to create a github release
+      pull-requests: write  # for release-drafter/release-drafter to add label to PR
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Release
```
Comment (Alxandr) [inline .github/workflows/create-release-draft.yml:20]:
```suggestion
      - name: Create release draft
```

File: .github/workflows/deploy-after-release.yml [hunk 15]
```diff
@@ -0,0 +1,119 @@
...
+        with:
+          registry: ${{ env.REGISTRY }}
+          repository: '${{ env.REPOSITORY }}'
+          target: ${{ github.sha }}
+          tags: ${{ github.event.release.tag_name }}
+
+  deploy-prod:
+    environment: PROD
+    runs-on: ubuntu-latest
+    needs: deploy-tt02
```
Comment (Alxandr) [inline .github/workflows/deploy-after-release.yml:90]:
```suggestion
    needs:
      - build
      - deploy-tt02

```

File: .github/workflows/deploy-after-release.yml [hunk 16]
```diff
@@ -0,0 +1,119 @@
...
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      - uses: ./.github/actions/deploy
+        name: Deploy
+        with:
+          image-tag: ${{ github.sha }}
+          resource-group: ${{ vars.CONTAINER_APP_RESOURCE_GROUP_NAME }}
+          container-app: ${{ vars.CONTAINER_APP_NAME }}
+          function-app: ${{ vars.AZURE_FUNCTIONAPP_NAME }}
+          function-app-path: ./artifacts/function-app
```
Comment (Alxandr) [inline .github/workflows/deploy-after-release.yml:119]:
```suggestion
          function-app-path: ./artifacts/function-app

```

**Review by Alxandr — APPROVED** [review]

File: .github/workflows/deploy-after-release.yml (hunk 13 above)
Comment (acn-dgopa) [reply .github/workflows/deploy-after-release.yml:79]: so tag before deploy?

File: .github/workflows/deploy-after-release.yml (hunk 16 above)
Comment (acn-dgopa) [reply .github/workflows/deploy-after-release.yml:119]: dont understand what is the change. I see that both lines are same

**Pull request merged**